src/jit/morph.cpp

```diff
@@ -13665,7 +13665,7 @@ GenTree* Compiler::fgMorphSmpOpOptional(GenTreeOp* tree)
     GenTree*   op2  = tree->gtOp2;
     var_types  typ  = tree->TypeGet();
 
-    if (GenTree::OperIsCommutative(oper))
+    if (!optValnumCSE_phase && GenTree::OperIsCommutative(oper))
```

Comment:
We could change the reordering optimization to only run in the first global morph phase.